tests/test_local.py

```diff
@@ -76,20 +78,30 @@ def test_reset_default_storage_dir(client_class, monkeypatch):
     cloud_prefix = client_class._cloud_meta.path_class.cloud_prefix
 
     p1 = client_class().CloudPath(f"{cloud_prefix}drive/file.txt")
-    client_class.reset_default_storage_dir()
-    p2 = client_class().CloudPath(f"{cloud_prefix}drive/file.txt")
-
     assert not p1.exists()
-    assert not p2.exists()
```

Comment:
This test fails with my new implementation, but I think the old behavior is not desirable. In the old version of this test, we reset the storage directory before instantiating `p2` and writing out `p1`, which means `p2` uses a new default local storage but `p1` still uses the old default local storage.